SwiftSampleProject/SwiftyVokoderTests/ManagedObjectContextTests.swift

```diff
@@ -0,0 +1,160 @@
+//
+//  ManagedObjectContextTests.swift
+//  SwiftSampleProject
+//
+//  Created by Carl Hill-Popper on 2/9/16.
+//  Copyright © 2016 Vokal.
+//
+
+import XCTest
+import Vokoder
+@testable import SwiftyVokoder
+
+class ManagedObjectContextTests: XCTestCase {
+
+    let manager = VOKCoreDataManager.sharedInstance()
+    
+    override func setUp() {
+        super.setUp()
+        self.manager.resetCoreData()
+        self.manager.setResource("CoreDataModel", database: "CoreDataModel.sqlite")
+        
+        Stop.vok_import(CTAData.allStopDictionaries())
+        self.manager.saveMainContextAndWait()
+    }
+    
+    func testContextChain() {
+        let tempContext = self.manager.temporaryContext()
+        //temp context is a child of the main context
+        XCTAssertEqual(tempContext.parentContext, self.manager.managedObjectContext)
+        //main context has a private parent context
+        XCTAssertNotNil(self.manager.managedObjectContext.parentContext)
+    }
+    
+    func testDeletingObjectsOnTempContextGetsSavedToMainContext() {
+        //get a temp context, delete from temp, save to main, verify deleted on main
+        
+        let tempContext = self.manager.temporaryContext()
+        
+        let countOfStations = self.manager.countForClass(Station.self)
+        XCTAssert(countOfStations > 0)
+        
+        tempContext.performBlockAndWait {
+            self.manager.deleteAllObjectsOfClass(Station.self, context: tempContext)
+            self.manager.saveAndMergeWithMainContextAndWait(tempContext)
```

Comment:
As the README says, `writeToTemporaryContext` is more of a convenience method: it will handle creating and saving the context for you.  The `saveAndMergeWithMainContext...` methods are for when you've already created a temp context yourself and want to save and merge changes back to main context (and persistent store).  

The `saveAndMergeWithMainContext...` methods are safe to call from any queue (and described as such in the header docs) because all of the context stuff is wrapped in `performBlock...` calls (like the `hasChanges` call you asked about earlier).